packages/core-js/internals/set-intersection.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+var SetHelpers = require('../internals/set-helpers');
+var getSetRecord = require('../internals/get-set-record');
+var iterateSimple = require('../internals/iterate-simple');
+
+var Set = SetHelpers.Set;
+var aSet = SetHelpers.aSet;
+var add = SetHelpers.add;
+var has = SetHelpers.has;
+var size = SetHelpers.size;
+var forEach = SetHelpers.forEach;
+
+// `Set.prototype.intersection` method
+// https://github.com/tc39/proposal-set-methods
+module.exports = function intersection(other) {
+  var O = aSet(this);
+  var otherRec = getSetRecord(other);
+  var result = new Set();
+  if (size(O) <= otherRec.size) forEach(O, function (e) {
+    if (otherRec.includes(e)) add(result, e);
+  });
+  else iterateSimple(otherRec.getIterator(), function (e) {
+    if (has(O, e)) add(result, e);
+  });
```

Comment:
this doesn't seem like it ensures the order in `result` of items contained in `O` match their ordering in `O` - iow, `new Set([1, 2, 3]).intersection(new Set([3, 2]))` should result in `new Set([2, 3])`, but i think this algorithm results in `new Set([3, 2])`.

I'm trying to think of a way to achieve it that doesn't involve iterating over the entirety of `O` - I'd love to see what you come up with.

Reply:
Thanks, I saw the problem with tests of this method, but you showed that the problem is in the missed 7e step.

It seems that without adding iteration over `O` ensuring `O.[[SetData]]` order is impossible.

Reply:
That was my conclusion as well. Perhaps @bakkot has a better idea?

Reply:
I don't think it's possible to do efficiently in userland. It's possible in engines only because they have access to the actual underlying data structure.